theano/scan_module/scan_opt.py

```diff
@@ -205,7 +211,10 @@ def remove_constants_and_unused_inputs_scan(node):
 # This is a global opt for historical reason
 # It should be possible to change it to a local opt.
 class PushOutNonSeqScan(gof.Optimizer):
-
+    """
+    A global optimizer for pushing out the variables inside the scan that
+    are not used by the scan.
```

Comment:
I don't like this docstring. The computation can be used by other part of the inner scan graph. What about:

    A global optimizer for pushing out the variables inside the scan that depend only on non-sequence.